packages/frontend-2/server/api/v2/healthz/readiness.get.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  ApolloClient,
+  ApolloQueryResult,
+  gql,
+  HttpLink,
+  InMemoryCache,
+  NormalizedCacheObject
+} from '@apollo/client/core'
+
+type GraphQLClient = ApolloClient<NormalizedCacheObject>
+
+const serverInfoVersionQuery = gql`
+  query serverInfo {
+    version
+  }
+`
+
+export default defineEventHandler(() => {
+  // as part of creating the apollo client it checks that the server can be reached
+  return createApolloClient('apiOrigin') //FIXME there's some magic going on with the config that clearly isn't applied here (yet?)
+    .then(() => {
+      return {
+        status: 'ready'
+      }
+    })
+    .catch(() => {
+      throw createError({
+        statusCode: 500,
+        name: 'InternalServerError',
+        message:
+          'Frontend is unable to reach the Graphql server, or the server is unable to query the database',
+        statusMessage: 'Internal Server Error'
+      })
+    })
+})
+
+// TODO copied from packages/server/modules/cli/commands/download/commit.ts
+// most likely a better place to put the below
+const createApolloClient = async (origin: string): Promise<GraphQLClient> => {
```

Comment:
No need to copy this from anywhere, because the new frontend already has its own Apollo Client setup.

Try any of the following options to get the client:
1. `useApolloClient().client`
2. `const {$apollo} = useNuxtApp()`

Now these will likely not work, because these API routes run in a very different context from the other app routes, but I want to test this out anyway.

If it doesn't work, then you'll have to create a new ApolloClient instance, but even then you can re-use the existing config, instead of setting everything up from scratch. `~~/lib/core/configs/apollo.ts` exports a function for building this config, so use that.

I do see that this function relies on some Vue/Nuxt composables like `useRuntimeConfig()` and `useAuthCookie()` which might not be available in API routes, but even in that case the function could just be split apart into a new one that allows you to feed in apiOrigin & authToken some other way. E.g.:

```ts

export const buildApolloConfig(params: {apiOrigin: string, authToken: Ref<Optional<string>>}) {
 ...
}

const defaultConfigResolver: APolloConfigResolver = async () => {
  const {
    public: { apiOrigin }
  } = useRuntimeConfig()

  const authToken = useAuthCookie()
  return buildApolloConfig({apiOrigin, authToken})
}

export default defaultConfigResolver;
```

And inside your API route:
```ts
import {ref} from 'vue'

const apiOrign = ... // still taken from runtime config somehow?
const authToken = ref(undefined); // i'm guessing we don't need a token here so we can leave it undefined
const config = await buildApolloConfig({apiOrigin, authToken});
```